### ISSUE TITLE
Handler with interface as generic type not working anymore

### ISSUE THREAD

**Issue by pdeparcq** [gautema/CQRSlite#51]
Hi,

I tried upgrading from v0.14.4 to v0.16.0 but the following event handler stopped working:

``` csharp
using System.Threading.Tasks;
using CQRSlite.Events;

namespace Televic.CoCon.Framework.Notification
{
    public class NotificationEventHandler : IEventHandler<IEvent>
    {
        private readonly Notifier _notifier;

        public NotificationEventHandler(Notifier notifier)
        {
            _notifier = notifier;
        }

        public Task Handle(IEvent message)
        {
            _notifier.Notify(message);

            return Task.CompletedTask;
        }
    }
}
```
All my other event handlers are still working, but this is a special case since I'm trying to handle all events here instead of events of a specific type.

**Comment by gautema** [comment]
Hi.

And thanks for the feedback. I'll look at this issue during the weekend. I haven't really thought about that use case, but I'll see what I can do.

**Comment by gautema** [comment]
One question, do you use the RouteRegistrar to register it to all events, or do you do it yourself?

**Comment by pdeparcq** [comment]
Hello,

Thanks for the nice framework by the way. I'm really impressed by it's power and simplicity.
Indeed, I'm using the RouteRegistrar to register my event handlers, but they are spread over 2 assemblies, so I'm doing something like this:

``` csharp
var bus = new RouteRegistrar(new DependencyResolver(_container));
bus.Register(ApplicationAssemblyType, typeof(NotificationEventHandler));
```

**Comment by gautema** [comment]
Hi. I released a preview last night 0.17.0-beta. Can you test this and check if it works for you?
